photutils/centroids/tests/test_core.py

```diff
@@ -82,24 +82,26 @@ def test_centroid_com_nan_withmask(use_mask):
         mask = np.zeros(data.shape, dtype=bool)
         mask[20, :] = True
         nwarn = 0
+        ctx = nullcontext()
     else:
         mask = None
         nwarn = 1
+        ctx = pytest.warns(AstropyUserWarning,
+                           match='Input data contains non-finite values')
 
-    with warnings.catch_warnings(record=True) as warnlist:
+    with ctx as warnlist:
         xc, yc = centroid_com(data, mask=mask)
-        assert_allclose(xc, xc_ref, rtol=0, atol=1.e-3)
-        assert yc > yc_ref
-    assert len(warnlist) == nwarn
+    assert_allclose(xc, xc_ref, rtol=0, atol=1.e-3)
+    assert yc > yc_ref
     if nwarn == 1:
-        assert issubclass(warnlist[0].category, AstropyUserWarning)
+        assert len(warnlist) == nwarn
 
-    with warnings.catch_warnings(record=True) as warnlist:
+    with pytest.warns(AstropyUserWarning,
```

Comment:
This warning always is emitted now that we don't use `warnings.catch_warnings`. You might want to double check if this is the intended behavior of photutils or a bug.